### PR TITLE
MAT-Merge

### PR DESCRIPTION
**Pull request by dtorvi**
Added merge.cpp, merge.hpp, and updated matUtils main.cpp

### CODE REVIEW

File: src/matUtils/merge.cpp [hunk 1]
```diff
@@ -0,0 +1,907 @@
...
+                    inp.best_node_num_leaves = &best_node_num_leaves;
+                    inp.best_set_difference = &best_set_difference;
+                    inp.best_node = &best_node;
+                    inp.best_j = &best_j;
+                    inp.num_best = &num_best;
+                    inp.j = k;
+                    inp.has_unique = &best_node_has_unique;
+                    inp.best_j_vec = &best_j_vec;
+                    inp.node_has_unique = &(node_has_unique);
+                    placement(inp, false, true);
```
Comment (yatisht) [inline src/matUtils/merge.cpp:295]: Why not just use mapper2_body?

Comment (dtorvi) [reply src/matUtils/merge.cpp:295]: I copied and pasted it in case I would have to modify it at all but I can change it so it just uses mapper2_body.

Comment (yatisht) [reply src/matUtils/merge.cpp:295]: Yes, let's not duplicate code.

**Comment by yatisht** [comment]
Looks good now. Please also add description of the merge command here: https://github.com/yatisht/usher/blob/e14305fca46bb55a3d8e9be349e4dd0ea81350c1/src/matUtils/main.cpp#L22 and add "[EXPERIMENTAL]" at the end. Thanks.

**Pull request merged**